learn/work/calibration-splits/index.qmd

```diff
@@ -0,0 +1,478 @@
+calibration(split_cal)
+```
+
+This is the basic idea for how we split sliding resamples. 
+
+Apart from the arguments to define the look forwards and backwards, `sliding_window()` has a few more options: `step`, `skip`, and `complete`. The options `step` and `skip` apply on the level of resamples, so we are not applying them to the (inner) split.  With `complete = FALSE`, `sliding_window()` allows you to create incomplete analysis sets (but not assessment sets).
+
+![](images/incomplete-window.jpg)
+
+Or, in code:
+
+```{r}
+#| label: sliding_window_incomplete
+
+df <- data.frame(x = 1:11)
+r_set <- sliding_window(
+  df,
+  lookback = 4,
+  assess_start = 3,
+  assess_stop = 5,
+  complete = FALSE
+)
+r_set
+```
+
+When creating the calibration split, we apply the `complete` option to the (now inner) analysis set and ensure that the calibration set is complete. In this example, it means that, until we have at least three rows (two for the complete calibration set, and at least one for the analysis set), we can't do a calibration split according to the rules laid out above. This is the case for the second resample, where we only have two rows to work with in the outer analysis set. Instead of erroring here, we fall back to not using calibration at all and return the (outer) analysis set as the (inner) analsysis set, along with an empty calibration set.
+
+```{r}
+#| label: sliding_window_split_incomplete_2
+
+r_split <- get_rsplit(r_set, 2)
+split_cal <- internal_calibration_split(r_split, .get_split_args(r_set))
+
+split_cal
+```
+
+A calibration split on the third resample, which contains three rows, succeeds with the required two rows in the calibration set and the remaining one in the incomplete analysis set.
+
+```{r}
+#| label: sliding_window_split_incomplete_3
+
+r_split <- get_rsplit(r_set, 3)
+split_cal <- internal_calibration_split(r_split, .get_split_args(r_set))
+split_cal
+```
+
+If we can fit a complete (inner) analysis set in the available rows, we do so, even if incomplete sets are allowed.
+
+```{r}
+#| label: sliding_window_split_incomplete_5
+
+r_split <- get_rsplit(r_set, 5)
+split_cal <- internal_calibration_split(r_split, .get_split_args(r_set))
+split_cal
+```
+
+
+### Index-based splitting
+
+When working with irregular series, it can be helpful to define resamples based on sliding over an index, rather than the rows directly. That index often is a form of time but it does not have to be. Let's take our previous data frame and use `x` as the index, with a small modification. The index doesn't go straight from 1 to 11 but rather is missing an observation at the index value of 2. We use the same  `lookback`, `assess_start`, and `assess_stop`. The difference is that now these define the resamples on the _index_ rather than the _rows_ directly.
+
+![](images/calibration-split-index.jpg)
+
+We still get two resamples, however, the analysis set contains only 4 rows because only those fall into the window defined by the index.
```

Comment:
Would it be beneficial to add the missing value at 1 or 5 such that the analysis sets have different length?

Reply:
or i guess, that isn't interesting at all. because then we do the same as the previous section